cmd/client/client.go

```diff
@@ -0,0 +1,94 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"os"
+)
+
+// RoomID represents the room ID
+type RoomID string
+
+type chatClient interface {
+	Run()
+	connectToRoom()
+	send()
+	recieve()
+}
+
+// Client - a struct for a client on the chat server
+type Client struct {
```

Comment:
Client isn't all about communication with server, what if I would ask you to add GUI to your chat?
I think you need to add a communicator struct&interface that would handle communication with server.
Client should include a communicator